File: exercise/perl/parsing_html.pl
#!/usr/bin/perl

use strict;
use warnings;

=head1 DESCRIPTION

This example script retrieves the html page from uniprot and extract
information from it.

dependencies:
perl-HTML-Tree

=cut

# print ASCII from HTML from a URL
use LWP::Simple;
use HTML::Parse;
my ($html, $ascii);
$html = get("http://www.uniprot.org/uniprot/Q42435");
defined $html
    or die "Can't fetch HTML from uniprot";

# Retrieve the protein name
$html =~ m/class="FULL">([\D\s]+?)<\/span>/;
my $name = $1;
print "Protein has for name: $name \n";

# Retrieve the protein short name
$html =~ m/Entry name<\/acronym><\/td><td>([\D\s]+?)<\/td>/;
my $shortname = $1;
print "Protein has for short-name: $shortname \n";

# Retrieve the protein's pathways
$html =~ m/Pathway<\/acronym><\/td><td>(.*?)<\/td>/;
my $pathway_string = $1;
my @pathways;
while ($pathway_string =~ m/">(.*?)<\/a>/g){
    push(@pathways, $1);
}

# Iterate through all the pathway found and print the information
foreach my $pathway(@pathways) {
    print "Protein has pathway: $pathway\n";
}

File: exercise/python/parsing_html.py
#!/usr/bin/python
#-*- coding: utf-8 -*-

"""
Parse uniprot's html to extract some information for the protein.
"""

import re
import requests

if __name__ == '__main__':
    # Load the information from uniprot
    html = requests.get('http://www.uniprot.org/uniprot/Q42435')

    # Retrieve the name of the protein
    m = re.search('class="FULL">([\D\s]+?)</span>', html.content)
    name = m.group(1)
    print 'Protein has for name         : %s' % name

    # Retrieve the short name of the protein
    m = re.search('Entry name</acronym></td><td>([\D\s]+?)</td>', html.content)
    short_name = m.group(1)
    print 'Protein has for short name   : %s' % short_name

    # Retrieve the list of pathways to which the protein is known to be
    # involved in
    m = re.search('Pathway</acronym></td><td>(.*?)</td>', html.content)
    pathways = re.findall('">(.*?)</a>', m.group(1))
    print 'Protein has for pathways     : %s' % ', '.join(pathways)
